dask/dataframe/io/tests/test_sql.py

```diff
@@ -288,7 +286,11 @@ def test_division_or_partition(db):
     m = out.map_partitions(
         lambda d: d.memory_usage(deep=True, index=True).sum()
     ).compute()
-    assert (50 < m).all() and (m < 200).all()
+    if pyarrow_strings_enabled():
+        lo, hi = 20, 100
+    else:
+        lo, hi = 50, 200
+    assert (m > lo).all() and (m < hi).all()
```

Comment:
I'm not sure this makes sense for this test. This is testing that `bytes_per_chunk=100` gives correctly sized chunks. In this case determining the chunk sizes probably needs to involve `pyarrow` strings earlier on in the stack.

Reply:
Doesn't this calculate memory usage per partition?

Reply:
I read through the test again, and now I see what you're saying. We expect memory usage per partition to not exceed `bytest_per_chunk` that we're passing in. Except at that point, the data has not been converted yet, and so we're overestimating how much memory it's going to take up. To properly fix that, `read_sql_table` needs to be aware of pyarrow strings.

I'm going to leave that test xfailed for now and fix it in a separate PR, since the fix is a bit more involved.